docs/source/developer/distributions.rst

```diff
@@ -0,0 +1,232 @@
+
+To use the ``metadist`` decorator, annotate the custom distribution class with ``@metadist``, passing in the attributes of the target distribution as parameters.
+
+For example, the following distributions use the decorator as follows:
+
+.. code-block:: python
+
+    @metadist(implements="core.multinoulli", var_type=["categorical", "discrete", "string"])
+    class MultinoulliDistribution(BaseDistribution):
+
+.. code-block:: python
+
+    @metadist(implements="core.unique_regex", var_type="string", is_unique=True)
+    class UniqueRegexDistribution(UniqueDistributionMixin, RegexDistribution):
+
+.. code-block:: python
+      
+    @metadist(implements="core.uniform_date", var_type="date")
+    class UniformDateDistribution(BaseUniformDistribution):
+
+
+The ``metadist`` decorator is implemented automatically as part of the main ``metasyn`` package. 
+
+
+Categorical submodule
+~~~~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.categorical` module contain the :class:`metasyn.distribution.categorical.MultinoulliDistribution` class, which is used for categorical distributions.
+
+Continuous submodule
+~~~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.continuous` module contains the classes used for continuous distributions.
+
+DateTime submodule
+~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.datetime` module contains the classes used for ``DateTime`` distributions.
+
+Discrete submodule
+~~~~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.discrete` module contains the classes used to for discrete distributions.
+
+Faker submodule
+~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.faker` module contains the classes used for distributions that are based on the `Faker <https://faker.readthedocs.io/en/master/>`_ package.
+
+NA submodule
+~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.na` module contains the :class:`metasyn.distribution.NADistribution` class, a distribution which generates *only* NA values.
+
+Regex submodule
+~~~~~~~~~~~~~~~
+The :mod:`~metasyn.distribution.regex` module contains the classes for distributions that are based on regular expressions. It implements the open source `regexmodel <https://github.com/sodascience/regexmodel>`_ package. 
+
+There is also a legacy module :mod:`~metasyn.distribution.legacy.regex` that contains the old implementation of the regex distribution. 
+
+
+Creating a new distribution
+---------------------------
+
+New distributions can be created by either directly inheriting from a base class such as the :class:`~metasyn.distribution.BaseDistribution` (or a specialized base class such as :class:`~metasyn.distribution.ScipyDistribution`)
+
+For example, let's say we want to implement a new series of similar distributions, we can create a base class inheriting from :class:`~metasyn.distribution.BaseDistribution`:
+
+.. code-block:: python
+
+    from metasyn.distribution.base import BaseDistribution, metadist
```

Comment:
I personally wouldn't really advise this in the developer's manual. Obviously it is possible and it is done in many places in the code, but I would just suggest creating a new distribution from one of the already existing base classes. If they want to create a new base class, be my guest, but they probably already know what they're doing anyway.